halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/tasks/v1/TaskRepository.java

```diff
@@ -20,83 +19,98 @@
  */
 @Slf4j
 public class TaskRepository {
-  static final Map<String, DaemonTaskStatus> tasks = new ConcurrentHashMap<>();
+  static final Map<String, DaemonTask> tasks = new ConcurrentHashMap<>();
+  static final Map<String, ShallowTaskInfo> namedTasks = new ConcurrentHashMap<>();
 
-  static public List<String> getTasks() {
-    return new ArrayList<>(tasks.keySet());
+  static public Map<String, ShallowTaskInfo> getTasks() {
+    return namedTasks;
+  }
+
+  static private void deleteTaskInfo(String uuid) {
+    tasks.remove(uuid);
+    namedTasks.remove(uuid);
   }
 
   static public <C, T> DaemonTask<C, T> submitTask(Supplier<DaemonResponse<T>> runner, String name) {
     DaemonTask<C, T> task = new DaemonTask<>(name);
     String uuid = task.getUuid();
     log.info("Scheduling task " + task);
     Runnable r = () -> {
+      Exception fatalError = null;
+      State state = null;
+      DaemonResponse<T> response = null;
       log.info("Starting task " + task);
       DaemonTaskHandler.setTask(task);
       task.setState(State.RUNNING);
       try {
-        DaemonResponse<T> response = runner.get();
-        task.setResponse(response);
-        task.setState(State.SUCCESS);
+        response = runner.get();
+        state = State.SUCCESS;
       } catch (HalException e) {
         log.info("Task " + task + " failed for reason: ", e);
-        task.setResponse(new DaemonResponse<>(null, new ProblemSet(e.getProblems())));
-        task.setFatalError(e);
-        task.setState(State.FATAL);
+        response = new DaemonResponse<>(null, new ProblemSet(e.getProblems()));
+        fatalError = e;
+        state = State.FATAL;
+      } catch (DaemonTaskInterrupted e) {
+        log.info("Task " + task + " interrupted: ", e);
```

Comment:
Both just set a few variables, there isn't much to extract that doesn't require global vars or references